nvm.sh

```diff
@@ -2420,13 +2420,26 @@ nvm() {
         nvm_err "wget: not found"
       fi
 
-      for tool in git grep awk sed cut basename rm mkdir xargs; do
+      local TEST_TOOLS
+      TEST_TOOLS="git grep awk"
+      if [ "linux" = "$(nvm_get_os)" ]; then
+        TEST_TOOLS="${TEST_TOOLS} sed cut basename rm mkdir xargs"
```

Comment:
why would generic "linux" (but not the others) want to test sed/cut/basename/rm/mkdir/xargs?

Reply:
This test also tests if the commands are aliased, built-in, or not, and the commands can come from GNU utils or BusyBox, or even BSD utils. So this test tells where are they from, are they aliased, and the versioning, to help debug the different behavior between versions.

Reply:
Right, but couldn't they be aliased on darwin or aix too, for example?

Reply:
didn't test them all because it'll be very complex to determine to get the version info or not, any suggestions please?

Reply:
These are posix tools, so "getting the version" should be the same in the majority case, no?

Reply:
As we discussed above, tools on macOS doesn't work with that ;(

Reply:
ok - so should the check be "!= darwin" instead of "= linux"?

Reply:
@ljharb any new idea here? Thanks.

Reply:
Since currently the logic is "everything", and we know it doesn't work on darwin, let's only branch on "not darwin" for now.

Reply:
We know it doesn't work on FreeBSD, so I'll add both.